media-libs/dav1d/dav1d-0.5.2.recipe

```diff
@@ -0,0 +1,82 @@
+SUMMARY="A new AV1 cross-platform decoder"
+DESCRIPTION="dav1d is a new AV1 cross-platform decoder, open-source, \
+and focused on speed and correctness. It supports all features from AV1, \
+including all subsampling and bit-depth parameters."
+HOMEPAGE="https://code.videolan.org/videolan/dav1d"
+COPYRIGHT="2018-2019, VideoLAN and dav1d authors"
+LICENSE="BSD (2-clause)"
+REVISION="1"
+SOURCE_URI="https://code.videolan.org/videolan/dav1d/-/archive/$portVersion/dav1d-$portVersion.tar.bz2"
+CHECKSUM_SHA256="1780199d71e62a82ca1bfca2d71da5c599c7e1edc68c72f6fc3c0c25ce1880b7"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64 ?arm ?ppc"
+
+libVersion="3.1.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	dav1d = $portVersion
+	cmd:dav1d
+	lib:libdav1d = $libVersionCompat
+	"
+REQUIRES="
+	haiku
+	"
+
+PROVIDES_devel="
+	dav1d_devel = $portVersion
+	devel:libdav1d = $libVersionCompat
+	"
+REQUIRES_devel="
+	dav1d == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:doxygen
+	cmd:gcc
+	cmd:meson >= 0.47
+	cmd:nasm >= 2.13.02
+	cmd:ninja
+	"
+
+defineDebugInfoPackage dav1d \
+	$binDir/dav1d
+
+BUILD()
+{
+	rm -f -rf build
+	mkdir build && cd build
+
+	meson .. --prefix=$prefix
+
+	ninja $jobArgs
```

Comment:
Thanks. I'm new to meson, haven't seen these before.

Reply:
Still need a `mkdir` else it will complain:

> ERROR: Directory /sources/dav1d-0.5.2/build is neither a Meson build directory nor a project source directory.
> Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 1
> Error: Build has failed - stopping.
> ~>